app/controllers/api/regions_controller.rb

```diff
@@ -28,16 +28,19 @@ class RegionsController < BaseController
 
     def show
       data = ActiveSupport::JSON.decode(params[:q]).symbolize_keys
+      data[:session_ids] ||= []
 
-      FLOAT_Q_ATTRS.each { |key| data[key] = data[key].to_f if data.key?(key) }
-      INT_Q_ATTRS.each { |key| data[key] = data[key].to_i if data.key?(key) }
-
-      data[:time_from] = data[:time_from] || Time.new(2010).to_i
-      data[:time_to] = data[:time_to] || Time.new(2100).end_of_year.to_i
+      if (data[:session_ids] != [] && Session.find(data[:session_ids].first).fixed?)
```

Comment:
Have you thought if there's any way of avoiding the query? `Session.find(data[:session_ids].first).fixed?`

Reply:
I considered creating a new separate endpoint for fixed sessions, but I wasn't sure if that's a good idea. 
I guess I could also send a flag from the frontend like `data[:session_type]`?

Reply:
When it comes to methods there's a smell for cases with a "control parameter" that does different things depending on a flag
```
def call(flag)
  flag ? do_something : do_something_else
end
```
https://www.rubydoc.info/github/troessner/reek/Reek/Smells/ControlParameter

Not totally sure this is as valid when it comes to endpoint. Personally I'd consider two different endpoints. I think it's more explicit